worker/categorize.py

```diff
@@ -101,9 +101,8 @@ def categorize_item(self, title, text):
 
         # Build a graph and identify keyword clusters
         clustering = None
-
-        if len(assigned_articles) > 2:
-            semantic_graph = self._build_semantic_graph(assigned_articles)
+        semantic_graph = self._build_semantic_graph(assigned_articles)
+        if semantic_graph.edges():
```

Comment:
Looks great. Did you try running ```python manage.py feed_seed``` and letting the categorization part run for a little while and see if anything breaks?